metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_query.py

```diff
@@ -133,7 +138,7 @@ def get_all_tags_in_database_without_propagation(db_name: str) -> str:
         domain as "DOMAIN"
         FROM snowflake.account_usage.tag_references
         WHERE (object_database = '{db_name}' OR object_name = '{db_name}')
-        AND domain in ('DATABASE', 'SCHEMA', 'TABLE', 'COLUMN')
+        AND lower(domain) in {allowed_object_domains}
```

Comment:
I'm a little worried that this will be significantly slower. e.g. if `domain` has an index, we won't be able to use the index right? Because we're applying a transformation on the column. I understand wanting to reuse `SnowflakeObjectDomain` but I'd rather not potentially compromise performance just to get rid of magic strings

Reply:
Perhaps we can use `SnowflakeObjectDomain.____.value.capitalize()` instead?

Reply:
That's a valid concern and highly dependent on how snowflake handles this. We are using [snowflake account usage views ](https://docs.snowflake.com/en/sql-reference/account-usage#account-usage-views) here and snowflake does not have concept of indices but rather uses [clustering and micro-partitions](https://docs.snowflake.com/en/user-guide/tables-micro-partitions) that impact query performance. Let me look around how snowflake handles such case sensitive comparisons and if there is a better way.

Reply:
I am going to remove lower casing in query , assuming the worst case.